src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -48,17 +63,12 @@ internal static class DiagnosticHelper
                 throw new ArgumentNullException(nameof(descriptor));
             }
 
-            LocalizableString message;
-            if (messageArgs == null || messageArgs.Length == 0)
-            {
-                message = descriptor.MessageFormat;
-            }
-            else
+            if (effectiveSeverity == ReportDiagnostic.Suppress)
             {
-                message = new LocalizableStringWithArguments(descriptor.MessageFormat, messageArgs);
+                return s_suppressedDiagnostic;
```

Comment:
This is not correct. Even suppressed diagnostics need to be created with correct fields, such as ID, message, etc. Note that source suppressed diagnostics show up in the error list (not by default, but can be enabled by changing the filter on `Suppression State` column) and can also be unsuppressed by right clicking the entry in the error list and executing the `Remove Suppression` command.

Reply:
@mavasani I thought the `(DiagnosticSeverity)(-2)` will make the compiler discard the diagnostic completely in https://github.com/dotnet/roslyn/blob/f7634f7503665406e717d27bbac581b13a0c21b6/src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs#L60-L63

Reply:
I don't think we want that - suppressed diagnostics need to be present in the IDE diagnostic system for bunch of features around them to keep working.

Reply:
Can you point out what feature(s) need them?